src/pages/global-authorization/index.js

```diff
@@ -0,0 +1,200 @@
+import { Box, Button, Grid, Tooltip, Typography, IconButton } from '@mui/material'
+import Icon from 'src/@core/components/icon'
+import { useContext, useState } from 'react'
+import { useResourceQuery } from 'src/hooks/resource'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import Table from 'src/components/Shared/Table'
+import { formatDateDefault } from 'src/lib/date-helper'
+import ErrorWindow from 'src/components/Shared/ErrorWindow'
+
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useWindow } from 'src/windows'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { DataSets } from 'src/resources/DataSets'
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import ResourceGlobalForm from './forms/ResourceGlobalForm'
+import AccessLevelForm from './forms/AccessLevelForm'
+import FieldGlobalForm from './forms/FieldGlobalForm'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+
+const GlobalAuthorization = () => {
+  const { getRequest } = useContext(RequestsContext)
+  const { stack } = useWindow()
+  const [errorMessage, setErrorMessage] = useState(null)
+  const [moduleIdSelection, setModuleIdSelection] = useState(10) //check new way
+
+  async function fetchGridData() {
+    return await getRequest({
+      extension: SystemRepository.ModuleClassRES.qry,
+      parameters: `_filter=&_moduleId=10`
+    })
+  }
+
+  async function fetchWithFilter({ filters }) {
```

Comment:
Both functions  appear similar. Let's combine them into one. Also stick to ?? instead of || here. ?? is a null check. || is a falsy check (false values)